src/vs/workbench/contrib/mcp/browser/mcpWorkbenchService.ts

```diff
@@ -241,15 +245,40 @@ export class McpWorkbenchService extends Disposable implements IMcpWorkbenchServ
 	async install(server: IWorkbenchMcpServer): Promise<void> {
 		if (server.installable) {
 			await this.mcpManagementService.install(server.installable);
-			return;
-		}
-
-		if (server.gallery) {
+		} else if (server.gallery) {
 			await this.mcpManagementService.installFromGallery(server.gallery, { packageType: server.gallery.packageTypes[0] });
-			return;
+		} else {
+			throw new Error('No installable server found');
 		}
 
-		throw new Error('No installable server found');
+		// After successful installation, check if the server has a readme and prompt
```

Comment:
This should be moved here - the action which user clicks

https://github.com/microsoft/vscode/blob/4b9018cbef9f9fca6c5a7b0ea9fb17b5e42b4390/src/vs/workbench/contrib/mcp/browser/mcpServerActions.ts#L127